src/vivarium_cluster_tools/psimulate/results/processing.py

```diff
@@ -7,50 +7,159 @@
 
 """
 
+import re
 import time
+from collections import defaultdict
 from pathlib import Path
 
 import pandas as pd
+import pyarrow.parquet as pq
 from loguru import logger
 
 from vivarium_cluster_tools import utilities as vct_utils
 from vivarium_cluster_tools.psimulate.paths import OutputPaths
 
+DEFAULT_BYTES_PER_ROW_ESTIMATE = 500.0
+
+
+class ChunkMap:
+    """Tracks current chunk number per metric during results writing.
+
+    This is mutable state that gets updated in place as chunks rotate.
+
+    Attributes
+    ----------
+    results_dir
+        Directory containing metric subdirectories with chunk files.
+    metrics
+        Dictionary mapping metric name to current chunk number. Defaults to 0 for new metrics.
+    """
+
+    def __init__(self, results_dir: Path, metrics: dict[str, int] | None = None) -> None:
+        self.results_dir = results_dir
+        self.metrics: defaultdict[str, int] = defaultdict(int)
+        if metrics is not None:
+            self.metrics.update(metrics)
+
+    @classmethod
+    def from_existing_results(cls, results_dir: Path) -> "ChunkMap":
+        """Create ChunkMap by scanning existing chunk files.
+
+        Parameters
+        ----------
+        results_dir
+            Directory containing metric subdirectories with chunk files.
+
+        Returns
+        -------
+            ChunkMap initialized with current chunk numbers per metric.
+        """
+        metrics: dict[str, int] = {}
+
+        if not results_dir.exists():
+            return cls(results_dir, metrics)
+
+        for metric_dir in results_dir.iterdir():
+            if not metric_dir.is_dir():
+                continue
+
+            chunk_files = list(metric_dir.glob("chunk_*.parquet"))
+            if not chunk_files:
+                continue
+
+            # Find highest chunk number
+            chunk_nums = []
+            for f in chunk_files:
+                match = re.match(r"chunk_(\d+)\.parquet", f.name)
```

Comment:
The way you've named these chunks, if you sort the filenames alphabetically, isn't the last one guaranteed to have the highest chunk number? And can't you use glob to provide a pattern that matches your expected filename pattern to filter out any extraneous files?